SimulationAgent/DeviceState/DeviceStateActor.cs

```diff
@@ -152,6 +153,10 @@ private ISmartDictionary GetInitialProperties(DeviceModel model)
         {
             var properties = new SmartDictionary();
 
+            // Add SupportedMethods property with methods listed in device model
+            properties.Set(SUPPORTED_METHODS_KEY, string.Join(",", this.deviceModel.CloudToDeviceMethods.Keys));
```

Comment:
Yay!

Reply:
Who is using this?  Is it for the UI to enumerate which methods are supported for devices?

Reply:
This seems like it should expose everything you need to describe to the client the function: name of function, parameters, return value, etc.  The UI will need to show a combo box for function name but then it's going to need to build UI for parameters at least isn't it?

Reply:
I'd picture  JSON schema that we put in the twin that offers what you need - suggest talking with @dluc about device models.

Reply:
This is copied directly from: https://github.com/Azure/device-simulation-dotnet/blob/0d37f326717db065dab767105763d29a0ece24bd/SimulationAgent/Simulation/DeviceStatusLogic/DeviceBootstrap.cs#L140



Added a new issue to track the new feature request:
https://github.com/Azure/device-simulation-dotnet/issues/191